PerfTools/AllocMonitor/plugins/moduleAlloc_setupFile.cc

```diff
@@ -515,6 +519,16 @@ namespace edm::service::moduleAlloc {
         esModuleLabelsPtr.reset();
         esModuleTypesPtr.reset();
       }
+      {
+        std::ostringstream oss;
+        if (!sourceTypesPtr->empty()) {
+          std::vector<std::string> sourceTypeList{1, *sourceTypesPtr};
+          std::vector<std::string> sourceNames{1, "source"};
+          moduleIdToLabelAndType(oss, sourceNames, sourceTypeList, 'S', "Source ID", "Source label", "Source type");
+          logFile->write(oss.str());
+          sourceTypesPtr.reset();
+        }
```

Comment:
How about writing the line unconditionally, but have a default value of e.g. `Unknown` in case the `watchPreSourceConstruction` somehow would not be called to set the value? You could set the default value already in `sourceTypesPtr` construction.